### PR TITLE
[FEAT] Piece with PieceType and PieceColor enums

### PR DESCRIPTION
**Pull request by MatthewLiuXu**
## Description                                                                                                                                                             
                                                                 
  Implement the abstract `Piece` class and its supporting enums (`PieceColor`, `PieceType`) as the foundation for the chess piece hierarchy.                                 
   
  ## Type of Change                                                                                                                                                          
                                                                 
  - [x] Feature (new functionality)
  - [ ] Bug fix
  - [x] Documentation update
  - [ ] Refactor                                                                                                                                                             
  - [ ] Test addition/update
                                                                                                                                                                             
  ## Related Work                                                

  Related to `docs/design/chess-full-design.puml`

  ## Changes Made

  - [x] Created `PieceColor` enum (WHITE, BLACK)
  - [x] Created `PieceType` enum (PAWN, ROOK, KNIGHT, BISHOP, QUEEN, KING)
  - [x] Created abstract `Piece` class with fields (`type`, `color`, `canJump`, `hasMoved`), getters, state management (`changeToMoved`, `resetHasMoved`), and abstract      
  methods (`makeCopy`)                                                                                                                                                       
  - [x] Added `hasMoved()` getter to design doc to reflect implementation                                                                                                    
                                                                                                                                                                             
  ## BVA & Testing                                               

  - BVA documented in: N/A (abstract class, no boundary conditions to analyze)                                                                                               
  - Test cases implemented: N/A (no concrete subclasses yet to instantiate)
  - All tests passing: N/A (no test sources; build succeeds)                                                                                                                 
                                                                 
  ## Design Alignment

  - [x] Changes align with `docs/design/chess-full-design.puml`                                                                                                              
  - [x] No breaking changes to existing methods
  - [x] New methods follow the established patterns                                                                                                                          
  - Note: `isValidMoveShape(Location, Location)` is omitted from this PR since `Location` is not yet implemented and movement logic is out of scope. `hasMoved()` was added
  to the design doc to match the implementation.                                                                                                                             
   
  ## Implementation Notes                                                                                                                                                    
                                                                 
  - `canJump` is derived in the `Piece` constructor from `type == PieceType.KNIGHT` rather than passed as a parameter, keeping the constructor signature matching the design 
  doc.
  - `hasMoved()` getter was added (not in original design doc) because subclasses need to read the private `hasMoved` field in their `makeCopy()` implementations. The design
   doc has been updated accordingly.                             

  ## Checklist

  - [x] Code compiles and all tests pass                                                                                                                                     
  - [x] No new compiler warnings                                 
  - [x] Documentation updated (if applicable)
  - [x] Commit messages are clear and follow TDD workflow

### CODE REVIEW

**Review by didiermun — COMMENTED** [review]
looks good!

File: src/main/java/domain/piece/Piece.java [hunk 1]
```diff
@@ -17,4 +17,28 @@ public Piece(PieceType type, PieceColor color) {
     public PieceType getType() {
         return type;
     }
+
```
Comment (didiermun) [inline src/main/java/domain/piece/Piece.java:20]: Thanks @MatthewLiuXu🙏 time to get to work!

**Review by didiermun — APPROVED** [review]

**Review by Alex-Anca — APPROVED** [review]
Acceptable.
